### DISCUSSION TITLE
vercel now deprecated pkg officially

### DISCUSSION THREAD

**Discussion by julianpoemp** [yao-pkg/pkg#19] [General]
As I found out, vercel deprecated pkg on January 3th 2024.

<img width="886" alt="Bildschirmfoto 2024-01-09 um 12 33 56" src="https://github.com/yao-pkg/pkg/assets/25926588/aa26d731-c6f0-489c-90ea-a77e39633499">

Still I don't understand why they stopped developing pkg. I think it's still a very important tool without alternatives that support all features from pkg. A lot of people are using pkg (that repository has over 24k stars).

As they stated they are looking forward the [new executable feature of NodeJS 21](https://nodejs.org/api/single-executable-applications.html). As far as I understand this still requires a NodeJS environment installed on the target machine. So it won't be an equal alternative to pkg.

What do you think about this?

**Comment by robertsLando** [comment]
> As they stated they are looking forward the [new executable feature of NodeJS 21](https://nodejs.org/api/single-executable-applications.html). As far as I understand this still requires a NodeJS environment installed on the target machine. So it won't be an equal alternative to pkg.

It is very far from be a drop in replacement of pkg, I'm following updates on it, here are some links:

- https://github.com/nodejs/single-executable/discussions/78#discussioncomment-7406623
- https://github.com/nodejs/single-executable/issues/83
- https://github.com/nodejs/single-executable/issues/68

It will be the future but in the meanwhile I will try to keep pkg updated with the latest nodejs versions. As I said in the first link the hardest part of keeping pkg updated is related to nodejs patches that requires lot of manual work and very few users are able to work with them when something chages on nodejs source code (me neighter). Also when working with packaged applications there are lot of edge cases where something could be broken when executed in a packaged app (see many issues opened on vercel/pkg).

IMO it will require many years until node-sea will be able to support everything needed and even then there will be the need to develop a package on top of it in order to make it work so easy like pkg
